idp/openid/openid-connect.go

```diff
@@ -66,6 +68,13 @@ type IdentityCreator interface {
 	CreateIdentity(context.Context, *oauth2.Token) (store.Identity, error)
 }
 
+// A GroupsRetriever is used to retrieve a list of user groups from the
+// OpenID token returned by the OpenID authentication process.
+type GroupsRetriever interface {
+	// Retrieve groups from the OpenID token.
```

Comment:
It's a bit silly, but usually godocs of functions are written with the name of the function as the first word: "RetrieveGroups retrieves groups from the OpenID token".